r/tests/testthat/test-compute-sort.R

```diff
@@ -0,0 +1,157 @@
+    sort(tbl$int, na.last = NA)
+  )
+  expect_equal(
+    as.vector(sort(Array$create(tbl$int), na.last = TRUE)),
+    sort(tbl$int, na.last = TRUE)
+  )
+  expect_equal(
+    as.vector(sort(Array$create(tbl$int), na.last = FALSE)),
+    sort(tbl$int, na.last = FALSE)
+  )
+  expect_equal(
+    as.vector(sort(Array$create(tbl$int), decreasing = TRUE)),
+    sort(tbl$int, decreasing = TRUE)
+  )
+  expect_equal(
+    as.vector(sort(Array$create(tbl$int), decreasing = TRUE, na.last = TRUE)),
+    sort(tbl$int, decreasing = TRUE, na.last = TRUE)
+  )
+  expect_equal(
+    as.vector(sort(Array$create(tbl$int), decreasing = TRUE, na.last = FALSE)),
+    sort(tbl$int, decreasing = TRUE, na.last = FALSE)
+  )
+})
+
+test_that("ChunkedArray sort on integers", {
+  expect_equal(
+    ChunkedArray$create(tbl$int[1:5], tbl$int[6:10])$SortIndices(),
+    Array$create(0L:9L, type = uint64())
+  )
+  expect_equal(
+    ChunkedArray$create(rev(tbl$int)[1:5], rev(tbl$int)[6:10])$SortIndices(descending = TRUE),
+    Array$create(c(1L:9L, 0L), type = uint64())
+  )
+  expect_equal(
+    as.vector(sort(ChunkedArray$create(tbl$int[1:5], tbl$int[6:10]))),
+    sort(tbl$int)
+  )
+  expect_equal(
+    as.vector(sort(ChunkedArray$create(tbl$int[1:5], tbl$int[6:10]), na.last = NA)),
+    sort(tbl$int, na.last = NA)
+  )
+  expect_equal(
+    as.vector(sort(ChunkedArray$create(tbl$int[1:5], tbl$int[6:10]), na.last = TRUE)),
+    sort(tbl$int, na.last = TRUE)
+  )
+  expect_equal(
+    as.vector(sort(ChunkedArray$create(tbl$int[1:5], tbl$int[6:10]), na.last = FALSE)),
+    sort(tbl$int, na.last = FALSE)
+  )
+  expect_equal(
+    as.vector(sort(ChunkedArray$create(tbl$int[1:5], tbl$int[6:10]), decreasing = TRUE)),
+    sort(tbl$int, decreasing = TRUE)
+  )
+  expect_equal(
+    as.vector(sort(ChunkedArray$create(tbl$int[1:5], tbl$int[6:10]), decreasing = TRUE, na.last = TRUE)),
+    sort(tbl$int, decreasing = TRUE, na.last = TRUE)
+  )
+  expect_equal(
+    as.vector(sort(ChunkedArray$create(tbl$int[1:5], tbl$int[6:10]), decreasing = TRUE, na.last = FALSE)),
+    sort(tbl$int, decreasing = TRUE, na.last = FALSE)
+  )
+})
+
+test_that("Array/ChunkedArray sort on strings", {
+  expect_equal(
+    as.vector(sort(Array$create(tbl$chr), decreasing = TRUE, na.last = FALSE)),
+    sort(tbl$chr, decreasing = TRUE, na.last = FALSE)
+  )
+  expect_equal(
+    as.vector(sort(ChunkedArray$create(tbl$chr[1:5], tbl$chr[6:10]), decreasing = TRUE, na.last = FALSE)),
+    sort(tbl$chr, decreasing = TRUE, na.last = FALSE)
+  )
+})
+
+test_that("Array/ChunkedArray sort on floats", {
+  skip("ARROW-12055")
+  expect_equal(
+    as.vector(sort(Array$create(tbl$dbl), decreasing = TRUE, na.last = FALSE)),
+    sort(tbl$dbl, decreasing = TRUE, na.last = FALSE)
+  )
+  expect_equal(
+    as.vector(sort(ChunkedArray$create(tbl$dbl[1:5], tbl$dbl[6:10]), decreasing = TRUE, na.last = FALSE)),
+    sort(tbl$dbl, decreasing = TRUE, na.last = FALSE)
+  )
+})
+
+test_that("Table/RecordBatch sort", {
```

Comment:
These tests aren't super readable

Reply:
I cleaned up the tests in adb45acc7e03c9dd74af05d7b56536aad0a0cf44. They now use the helper function `expect_dplyr_equal()` that I added in ddaffcd41b899a79f95f08717544db3b3266d273 which improves readability.